posthog/api/cohort.py

```diff
@@ -141,15 +131,28 @@ def _calculate_static_by_csv(self, file, cohort: Cohort) -> None:
         distinct_ids_and_emails = [row[0] for row in reader if len(row) > 0 and row]
         calculate_cohort_from_list.delay(cohort.pk, distinct_ids_and_emails)
 
+    def _validate_filters(self, validated_data: Dict):
```

Comment:
I still added more parsing to make sure we don't save junk (or atleast it's in Dict form) we can probably do even more here to enforce what's saved into cohort.filters

Reply:
hmm, why not just call this `validate_filters`, so its run automagically for both creates and updates? ( https://docs.djangoproject.com/en/4.0/ref/validators/ ).

And, should we allow strings here at all? Doesn't make sense to me? I'd say we should reject anything that is not a dict with the `properties` key ?